src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -139,5 +147,58 @@ public TemplateCache(JObject? contentJobject, ILogger logger)
             while (currentCulture.Name != CultureInfo.InvariantCulture.Name);
             return null;
         }
+
+        private (string, JObject?)? GetBestHostConfigMatch(IScanTemplateInfo newTemplate, IEngineEnvironmentSettings settings, IMountPoint mountPoint)
+        {
+            if (newTemplate.HostConfigFiles.TryGetValue(settings.Host.HostIdentifier, out string? preferredHostFilePath))
+            {
+                return (preferredHostFilePath, ReadHostFile(newTemplate, preferredHostFilePath, settings, mountPoint));
+            }
+
+            foreach (string fallbackHostName in settings.Host.FallbackHostTemplateConfigNames)
+            {
+                if (newTemplate.HostConfigFiles.TryGetValue(fallbackHostName, out string? fallbackHostFilePAth))
+                {
+                    return (preferredHostFilePath, ReadHostFile(newTemplate, fallbackHostFilePAth, settings, mountPoint));
+                }
+            }
+            return null;
+        }
+
+        private JObject? ReadHostFile(IScanTemplateInfo template, string path, IEngineEnvironmentSettings settings, IMountPoint mountPoint)
+        {
+            if (template.GeneratorId != RunnableProjectGeneratorId)
+            {
+                return null;
+            }
+            settings.Host.Logger.LogDebug($"Start loading host config {template.MountPointUri}{path}");
+            try
+            {
+                IFile? hostFile = mountPoint.FileInfo(path);
+                if (hostFile == null || !hostFile.Exists)
+                {
+                    throw new FileNotFoundException($"Host file {hostFile?.GetDisplayPath()} does not exist.");
+                }
+                using (var sr = new StreamReader(hostFile.OpenRead()))
+                using (var jsonTextReader = new JsonTextReader(sr))
```

Comment:
nit: shorthand `using` equivalent (without block) will lead to shorter code here

Reply:
used `JExtensions` instead